src/test/java/cubicchunks/world/column/TestCubeMap.java

```diff
@@ -0,0 +1,163 @@
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *  THE SOFTWARE.
+ */
+package cubicchunks.world.column;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.spongepowered.test.launch.LaunchWrapperTestRunner;
+
+import java.util.HashSet;
+import java.util.Random;
+import java.util.Set;
+
+import cubicchunks.world.cube.Cube;
+
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(LaunchWrapperTestRunner.class)
+public class TestCubeMap {
+
+	@Test
+	public void addCubesInOrder() {
+		CubeMap testedCubeMap = new CubeMap();
+		for (int y = -30;y < 30;y++) {
+			testedCubeMap.put(createCube(y));
+		}
+
+		checkCubeMap(testedCubeMap, 60);
+
+		for (int y = -30;y < 30;y++) {
+			try {
+				testedCubeMap.put(createCube(y));
+				fail("Was able to add a Cube to the CubeMap at the same y coordinate as another Cube");
+			} catch (IllegalArgumentException e) {
+				// this should always happen
+			}
+		}
+	}
+
+	@Test
+	public void addCubesInReverse() {
+		CubeMap testedCubeMap = new CubeMap();
+		for (int y = 29;y >= -30;y--) {
+			testedCubeMap.put(createCube(y));
+		}
+
+		checkCubeMap(testedCubeMap, 60);
+
+		for (int y = -30;y < 30;y++) {
+			try {
+				testedCubeMap.put(createCube(y));
+				fail("Was able to add a Cube to the CubeMap at the same y coordinate as another Cube");
+			} catch (IllegalArgumentException e) {
+				// this should always happen
+			}
+		}
+	}
+
+	@Test
+	public void findRangeOfCubes() {
+		CubeMap testedCubeMap = new CubeMap();
+		for (int y = -30;y < 30;y++) {
+			testedCubeMap.put(createCube(y));
+		}
+
+		checkRange(testedCubeMap, 3, 3);
+
+		checkRange(testedCubeMap, -5, 5);
+		checkRange(testedCubeMap, -5, 500);
+		checkRange(testedCubeMap, -500, 5);
+		checkRange(testedCubeMap, -500, 500);
+
+		checkRange(testedCubeMap, 500, 500);
+		checkRange(testedCubeMap, -500, -500);
+
+		Random rang = new Random(12345);
+		testedCubeMap = new CubeMap();
+		for (int y = -40;y < 40;y++) {
+			if(rang.nextBoolean()) {
+				testedCubeMap.put(createCube(y));
+			}
+		}
+
+		checkRange(testedCubeMap, 3, 3);
+
+		checkRange(testedCubeMap, -5, 5);
+		checkRange(testedCubeMap, -5, 500);
+		checkRange(testedCubeMap, -500, 5);
+		checkRange(testedCubeMap, -500, 500);
+
+		checkRange(testedCubeMap, 500, 500);
+		checkRange(testedCubeMap, -500, -500);
+	}
+
+	private void checkRange(CubeMap cubeMap, int lowerBound, int upperBound) {
+		Set<Cube> range = new HashSet<>();
+		cubeMap.cubes(lowerBound, upperBound).forEach(cube -> {
+			assertFalse(range.contains(cube)); // check for repeated cubes
```

Comment:
use `assertThat` where possible... it provides much nicer output when it fails.

Here it would probably be `assertThat(range, hasItem(cube))`